src/conditions/condition-expr.ts

```diff
@@ -0,0 +1,108 @@
+import { Conditions as WASMConditions } from '@nucypher/nucypher-core';
+import { ethers } from 'ethers';
+
+import { objectEquals, toJSON } from '../utils';
+
+import {
+  Condition,
+  ContractCondition,
+  RpcCondition,
+  TimeCondition,
+} from './base';
+import { BLOCKTIME_METHOD } from './base/time';
+import { CompoundCondition } from './compound-condition';
+import { ConditionContext } from './context';
+
+export type ConditionExpressionJSON = {
+  version: string;
+  condition: Record<string, unknown>;
+};
+
+export class ConditionExpression {
+  static VERSION = '1.0.0';
+
+  constructor(
+    public readonly condition: Condition,
+    public readonly version: string = ConditionExpression.VERSION
+  ) {}
+
+  public toObj(): ConditionExpressionJSON {
+    // TODO add version here
+    const conditionData = this.condition.toObj();
+    return {
+      version: this.version,
+      condition: conditionData,
+    };
+  }
+
+  public static fromObj(obj: ConditionExpressionJSON): ConditionExpression {
+    const version = obj.version;
+    // version specific logic can go here
+    const receivedMajorVersion = version.split('.')[0];
+    const currentMajorVersion = ConditionExpression.VERSION.split('.')[0];
```

Comment:
This is not an improvement, but I wanted to highlight an alternative way to access `split` results using deconstruction patterns.

```suggestion
    const [major, _minor, _fix] = version.split('.');
```

Reply:
Thanks. Looks like there is a `SemVer` object I can use - https://www.npmjs.com/package/semver. I'll try it out.